.pre-commit-config.yaml

```diff
@@ -0,0 +1,23 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+    - id: check-added-large-files
+    - id: check-yaml
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.1.5
+  hooks:
+    - id: ruff  # Run the linter
+      args: [ --fix ]
+      types_or: [ python, pyi, jupyter ]
+    - id: ruff-format  # Run the formatter
+      types_or: [ python, pyi, jupyter ]
+
+# https://pre-commit.ci/#configuration
+ci:
+  autofix_prs: true
+  autoupdate_schedule: quarterly
```

Comment:
Have set-up [pre-commit.ci](https://pre-commit.ci) GitHub App to fix lint/formatting errors in Pull Requests by default, but we can disable this if it gets annoying.

Also, `pre-commit autoupdate` will be ran quarterly (default is weekly, but that might get noisy).

Extra configs can be found at https://pre-commit.ci/#configuration